lib/surrealist/carrier.rb

```diff
@@ -10,21 +10,23 @@ class Carrier
     #   as instance's class name.
     # @param [Boolean] include_namespaces optional argument for having root key as a nested hash of
     #   instance's namespaces. Animal::Cat.new.surrealize -> (animal: { cat: { weight: '3 kilos' } })
+    # @param [String] root optional argument for using a specified root key for the resulting hash
     # @param [Integer] namespaces_nesting_level level of namespaces nesting.
     #
     # @raise ArgumentError if types of arguments are wrong.
     #
     # @return [Carrier] self if type checks were passed.
-    def self.call(camelize:, include_root:, include_namespaces:, namespaces_nesting_level:)
-      new(camelize, include_root, include_namespaces, namespaces_nesting_level).sanitize!
+    def self.call(camelize:, include_root:, include_namespaces:, root:, namespaces_nesting_level:)
+      new(camelize, include_root, include_namespaces, root, namespaces_nesting_level).sanitize!
     end
 
-    attr_reader :camelize, :include_root, :include_namespaces, :namespaces_nesting_level
+    attr_reader :camelize, :include_root, :include_namespaces, :root, :namespaces_nesting_level
 
-    def initialize(camelize, include_root, include_namespaces, namespaces_nesting_level)
+    def initialize(camelize, include_root, include_namespaces, root, namespaces_nesting_level)
       @camelize                 = camelize
       @include_root             = include_root
       @include_namespaces       = include_namespaces
+      @root                     = root
```

Comment:
We need to add sanitization of `root` parameter to `#sanitize!` method and specs for it (`spec/carrier_spec.rb`). Right now there are specs only for strings and nil. It should work with symbols as well (`Something.surrealize(root: :wrapper)`). I suggest that we raise `ArgumentError` if `root` is anything except non-empty string or a symbol (Class, Integer and so on).

Reply:
What about `nil`? Seems like the way the code path is setup, `Surrealist#call` is always expecting something for all of its arguments. In the case of `include_root`, for example, this is fine because it just defaults to false, which does nothing. But if only non-empty string and a symbol are allowed for `root`, then there is no way to prevent the schema from being wrapped into `root`.

Reply:
Yeah, good point. There are two ways here:
1. You can default `root` argument to `''` instead of `nil` and then validate it on not being anything except string or symbol.
2. You can leave default `root` argument as it is (`nil`), and then validate it on being either `nil`, non-empty string or a symbol.

I think I prefer the second option, it feels more naturally to have `nil` if we don't want to do anything. What do you think?

Reply:
I agree, `nil` seems more natural to me.

Reply:
@AlessandroMinali yes, you are right, we need to strip string argument. This is not relevant for other arguments, because they are either booleans or integers. @chrisatanasian could you please add a method in `Carrier.sanitize!` that will strip `root` argument?
And also add specs for this case:
`Animal.surrealize(root: '   wat   ') #=> '{ "wat" => {...} }'`

Reply:
Done.